app/models/course/assessment/answer.rb

```diff
@@ -79,8 +79,14 @@ def validate_consistent_grade
   #
   # @return [Course::Assessment::Answer::AutoGrading]
   def ensure_auto_grading!
-    create_auto_grading! unless auto_grading
+    new_auto_grading = auto_grading || create_auto_grading
+    if new_auto_grading.errors[:answer_id].empty?
```

Comment:
create_auto_grading! will throw an exception if validation fails. Places where record can be concurrently created:
 - between the auto_grading check and the create_auto_grading call
 - before validation, so validation fails
 - after validation, then AR will throw RecordNotUnique
 -